engine/kv_engine.cpp

```diff
@@ -623,56 +623,50 @@ Status KVEngine::Recovery() {
 
 Status KVEngine::HashGetImpl(const pmem::obj::string_view &key,
                              std::string *value, uint16_t type_mask) {
-  std::unique_ptr<DataEntry> data_entry(
+  std::unique_ptr<DataEntry> data_entry_meta(
```

Comment:
No need to use polymorphism if we just use Header and Meta when Hash Get.

Reply:
hash table search will copy DLDataentry if it's a sorted record